src/main/java/org/shanerx/tradeshop/listeners/CustomInventoryListener.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *     Copyright (c) 2016-2017 SparklingComet @ http://shanerx.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * NOTICE: All modifications made by others to the source code belong
+ * to the respective contributor. No contributor should be held liable for
+ * any damages of any kind, whether be material or moral, which were
+ * caused by their contribution(s) to the project. See the full License for more information
+ */
+
+package org.shanerx.tradeshop.listeners;
+
+import org.bukkit.ChatColor;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.inventory.Inventory;
+import org.shanerx.tradeshop.utils.Utils;
+
+public class CustomInventoryListener extends Utils implements Listener {
+
+	@EventHandler
+	public void onInventoryClick(InventoryClickEvent event) {
+
+		Inventory inventory = event.getInventory();
+
+		if (ChatColor.stripColor(inventory.getName()).endsWith("Shop                                 ")) {
```

Comment:
Beg pardon, but what is this?

Reply:
Prevents people from taking things out of our custom inventories, kinda necessary otherwise you could literally just pull the items we put in them out. Also there is a counted amount of spaces after the name so i know its our inventory, don't know how many it is anymore though. But that is the only way to distinguish between ours and others without having a terrible looking filled title on it.

Reply:
Alright, makes sense. Just got confused.
Would be more practical to have it defined as a constant somewhere, since I assume it appears elsewhere, where the title is set. Would make changing the identifier easier.